src/map/clif.cpp

```diff
@@ -4656,26 +4656,27 @@ void clif_leavechat(struct chat_data* cd, map_session_data* sd, bool flag)
 /// Opens a trade request window from char 'name'.
 /// 00e5 <nick>.24B (ZC_REQ_EXCHANGE_ITEM)
 /// 01f4 <nick>.24B <charid>.L <baselvl>.W (ZC_REQ_EXCHANGE_ITEM2)
-void clif_traderequest(map_session_data* sd, const char* name)
-{
-	int fd = sd->fd;
+void clif_traderequest(map_session_data& sd, const char* name){
 
-#if PACKETVER < 6
-	WFIFOHEAD(fd,packet_len(0xe5));
-	WFIFOW(fd,0) = 0xe5;
-	safestrncpy(WFIFOCP(fd,2), name, NAME_LENGTH);
-	WFIFOSET(fd,packet_len(0xe5));
-#else
-	map_session_data* tsd = map_id2sd(sd->trade_partner);
-	if( !tsd ) return;
+	PACKET_ZC_REQ_EXCHANGE_ITEM p{};
 
-	WFIFOHEAD(fd,packet_len(0x1f4));
-	WFIFOW(fd,0) = 0x1f4;
-	safestrncpy(WFIFOCP(fd,2), name, NAME_LENGTH);
-	WFIFOL(fd,26) = tsd->status.char_id;
-	WFIFOW(fd,30) = tsd->status.base_level;
-	WFIFOSET(fd,packet_len(0x1f4));
+	p.packetType = HEADER_ZC_REQ_EXCHANGE_ITEM;
+	safestrncpy(p.requesterName, name, sizeof(p.requesterName));
+#if PACKETVER > 6
+	// The client doesn't need this info to sucessfully trade, just to show info
+	map_session_data* tsd = map_id2sd(sd.trade_partner);
+
+	if( tsd != nullptr ){
```

Comment:
i tested it, 
		p.targetId = 0;
		p.targetLv = 0;
it's only to show info on tradebox
![image](https://github.com/user-attachments/assets/e8781569-d249-4787-8ea7-2f4f8ec8414f)

if it send 0 just don't show nv and PN info, but trade box still working fine

Reply:
don't worry because everything is checked before the cliff

![image](https://github.com/user-attachments/assets/5631611f-b9d8-4871-8ade-3db6f863418d)
![image](https://github.com/user-attachments/assets/25e0ffc5-9f26-499b-96d8-1fb8b345068e)

Reply:
so now, idk if it's need to fill with 0 because packetlen stay same

Reply:
i did some changes to stop casting mapid2sd here